### DISCUSSION TITLE
Does retsnoop work with debuginfod w/o vmlinux?

### DISCUSSION THREAD

**Discussion by aetimmes** [anakryiko/retsnoop#87] [Q&A] [answered]
I'm running a number of ARM Bottlerocket AMI images in EKS and have been using retsnoop to trace kernel functions.

There is no vmlinux image available as far as I know (only a stripped vmlinuz) and while there's no kernel-debug package, the Bottlerocket devs are discussing adding a debuginfod service. Would retsnoop be able to use this to trace inline functions?

**Comment by anakryiko** [comment, marked as answer]
Retsnoop cannot *trace* inline functions, but you are probably asking about stack symbolization of inline functions, right? I don't think I want to add debuginfod support to retsnoop itself, but we have plans to switch stack symbolization to https://github.com/libbpf/blazesym which does (or will) support debuginfod, once the transitions happens.

**Reply by aetimmes** [reply]
That makes sense - thank you!
